opentelemetry-sdk/src/opentelemetry/sdk/_metrics/export/__init__.py

```diff
@@ -77,3 +89,89 @@ def export(self, metrics: Sequence[Metric]) -> MetricExportResult:
 
     def shutdown(self) -> None:
         pass
+
+
+class PeriodicExportingMetricReader(MetricReader):
+    """`PeriodicExportingMetricReader` is an implementation of `MetricReader`
+    that collects metrics based on a user-configurable time interval, and passes the
+    metrics to the configured exporter.
+    """
+
+    def __init__(
+        self,
+        exporter: MetricExporter,
+        export_interval_millis: Optional[float] = None,
+        export_timeout_millis: Optional[float] = None,
+        preferred_temporality: AggregationTemporality = AggregationTemporality.CUMULATIVE,
+    ) -> None:
+        super().__init__(preferred_temporality=preferred_temporality)
+        self._exporter = exporter
+        if export_interval_millis is None:
+            try:
+                export_interval_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_INTERVAL", 60000)
+                )
+            except ValueError:
+                _logger.warning(
+                    "Found invalid value for export interval, using default"
+                )
+                export_interval_millis = 60000
+        if export_timeout_millis is None:
+            try:
+                export_timeout_millis = float(
+                    environ.get("OTEL_METRIC_EXPORT_TIMEOUT", 30000)
+                )
+            except ValueError:
+                _logger.warning(
+                    "Found invalid value for export timeout, using default"
+                )
+                export_timeout_millis = 30000
+        self._export_interval_millis = export_interval_millis
+        self._export_timeout_millis = export_timeout_millis
+        self._shutdown = False
+        self._daemon_thread = Thread(target=self._ticker, daemon=True)
+        self._condition = Condition()
+        self._lock = Lock()
+        self._daemon_thread.start()
+        if hasattr(os, "register_at_fork"):
+            os.register_at_fork(
+                after_in_child=self._at_fork_reinit
+            )  # pylint: disable=protected-access
+
+    def _at_fork_reinit(self):
+        self._condition = Condition()
+        self._lock = Lock()
+        self._daemon_thread = Thread(target=self._ticker, daemon=True)
+        self._daemon_thread.start()
+
+    def _ticker(self) -> None:
+        interval_secs = self._export_interval_millis / 1e3
+        while not self._shutdown:
+            with self._condition:
+                # might have received shutdown request
+                if not self._shutdown:
+                    self._condition.wait(interval_secs)
+                if self._shutdown:
+                    break
+            self.collect()
+        # one last collection below before shutting down completely
+        self.collect()
+
+    def _receive_metrics(self, metrics: Iterable[Metric]) -> None:
+        token = attach(set_value(_SUPPRESS_INSTRUMENTATION_KEY, True))
```

Comment:
But metrics don't have a feedback loop like SimpleSpanProcessor creates. I think it will behave more like BatchSpanProcessor in tracing. I could understand how maybe that's not desirable tho. Wdyt?

Reply:
> I could understand how maybe that's not desirable tho.

Yes, we want to avoid this. We already do this in Batch[Span|Log]Processor for same reason.

Reply:
Just want to call out that sometimes you may want self observability for an exporter, like gRPC/http latency for exporters. I suppose those exporters could always un-suppress the instrumentation so I think this default makes sense

Reply:
@aabmass is there any other change you want to have this conversation resolved? If we resolve this conversation we can merge this PR.

Reply:
🚢